service/ec2/customizations.go

```diff
@@ -8,7 +8,43 @@ import (
 	"github.com/aws/aws-sdk-go-v2/internal/awsutil"
 )
 
+type retryer struct {
+	aws.DefaultRetryer
+}
+
+func (d retryer) RetryRules(r *request.Request) time.Duration {
+	switch r.Operation.Name {
+	case opModifyNetworkInterfaceAttribute:
+		fallthrough
+	case opAssignPrivateIpAddresses:
+		return d.customRetryRule(r)
+	default:
+		return d.DefaultRetryer.RetryRules(r)
+	}
+}
+
+func (d retryer) customRetryRule(r *request.Request) time.Duration {
+	d.MinTime = 1000
+	return d.DefaultRetryer.RetryRules(r)
```

Comment:
Specifying the min time should be done when the custom retrier is created, instead of in the function call.